### ISSUE TITLE
OpenSUSE Tumbleweed: webkit2gtk-4.0 not found in package names

### ISSUE THREAD

**Issue by vcg3rd** [prologic/gopherclient#14]
On openSUSE Tumbleweed , make fails with `'webkit2gtk-4.0' not found in package names`

Searching packages I find `libwebkit2gtk-4_0-37` and `webkit2gtk-4_0-injected-bundles`

Seems to be a naming issue. Can I create a symbolic link to either of the second two and name it the first, so I don't break what uses the first?

**Comment by prologic** [comment]
Can you show me the full build output? I'm on macOS here so don't readily have an openSUSE system to test with (_I'll have to spin up a VM_).

You will have to install the development package I _think_ in order for the build system to link against webkit.

**Comment by vcg3rd** [comment]
```
`go: finding github.com/GeertJohan/go.rice/rice latest
# pkg-config --cflags  -- gtk+-3.0 webkit2gtk-4.0
Package gtk+-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtk+-3.0.pc'
to the PKG_CONFIG_PATH environment variable
Package 'gtk+-3.0', required by 'virtual:world', not found
Package 'webkit2gtk-4.0', required by 'virtual:world', not found
pkg-config: exit status 1
make: *** [Makefile:17: deps] Error 2
```

I missed the gtk error the first time, but if I issue the command `pkg-config --cflags gtk+-3.0
`
I get:

`-I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/freetype2 -I/usr/include/uuid -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/wayland -I/usr/include/libxkbcommon -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -pthread `

And I have gtk3-devel, gtk3-tools,gtk3-data, and gtk3-schema, and gtk3-branding-openSUSE - The GTK+ toolkit library (version 3) and more installed.

Don't knock yourself out spinning up a VM yet.  I'll see if I can figure it out. Thanks!

**Comment by vcg3rd** [comment]
OK, so that package `webkit2gtk-4_0-injected-bundles` had a -devel that wasn't installed, so I did.

Now make fails with:

`go: finding github.com/GeertJohan/go.rice/rice latest
# github.com/zserge/webview
In file included from webview.go:27,
                 from _cgo_export.c:4:
../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h: In function ‘external_message_received_cb’:
../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:253:3: warning: ‘webkit_javascript_result_get_global_context’ is deprecated [-Wdeprecated-declarations]
  253 |   JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
      |   ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
   49 | webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from webview.go:27,
                 from _cgo_export.c:4:
../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:254:3: warning: ‘webkit_javascript_result_get_value’ is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
  254 |   JSValueRef value = webkit_javascript_result_get_value(r);
      |   ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
   52 | webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# github.com/zserge/webview
In file included from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h: In function ‘external_message_received_cb’:
../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:253:3: warning: ‘webkit_javascript_result_get_global_context’ is deprecated [-Wdeprecated-declarations]
  253 |   JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
      |   ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
   49 | webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:254:3: warning: ‘webkit_javascript_result_get_value’ is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
  254 |   JSValueRef value = webkit_javascript_result_get_value(r);
      |   ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from ../../go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
   52 | webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: rice: Command not found
make: *** [Makefile:18: deps] Error 127
`

**Comment by prologic** [comment]
This:

```
make: rice: Command not found
```

Can you try re-running:

```#!sh
go get github.com/GeertJohan/go.rice/rice
```

This should have run from `make deps`

**Comment by vcg3rd** [comment]
> Can you try re-running:
> `go get github.com/GeertJohan/go.rice/rice`

I immediately went [here](https://github.com/GeertJohan/go.rice) and followed these instructions:
```
go get github.com/GeertJohan/go.rice
go get github.com/GeertJohan/go.rice/rice
```

And got the same fail on make.

After your reply, I decided to try a reboot.  Still no joy.

But I'm not used to this sort of responsiveness!  It's good, but it's making me feel guilty. :-) I love the look and features of your client, and I'm determined to use it even if I have to scrap openSuse, which I just installed yesterday, after not using it since before the Novell acquisition, and it's fighting me.  

Can't seem to make apt spell zypper. :-)

**Comment by prologic** [comment]
Haha You're welcome? I guess 🤣

I admit I've had a lot of issues trying to build this on Linux in general See #8 

I'd actually appreciate some help here as I have no access to any Linux Desktop systems at all besides spinning up a VirtualBox VM or two.

I just confirmed and pushed a few commits to master just now. Builds are working on my Mac at least where I install webkit from source via [brew](https://brew.sh)

**Comment by vcg3rd** [comment]
Well, I'm stuck for now.  Didn't notice you're on a Mac. I don't know any programming, let alone Go, but I thought maybe it was my $GOPATH, so I ensured it pointed to $HOME/go/bin where rice is.

I tried make as root. I tried your new master commit. I even backed up the Makefile and played around with import, which I doubt even belongs in a go Makefile, but what the heck...I like to tinker. Read a bit about rice embed-go, which seemed to be part of the build process, but you have it under deps: anyway, and I wouldn't know how to try to run rice except from its directory, and from there I wouldn't know what to do with it

I just can't seem to make it find rice, which is there.

I don't have anymore time today, but in main.go you're importing "path".  Is that $PATH or $GOPATH? And also from what I can tell you're importing rice `rice "github.com/GeertJohan/go.rice"`

Don't laugh, but I don't suppose import "github.com/GeertJohan/go.rice" without rice before it would matter, the way Johan describes it "Package usage Import the package: import "github.com/GeertJohan/go.rice" at that link?

I'll come back to it, just because I'm learning stuff, and there are never very many ways I can contribute, so if I can bumble around and help, I'm glad to bumble.

**Comment by prologic** [comment]
Ahh I know what you're missing here :)

Add this to your shell's settings:

```#!sh
export GOPATH="$HOME/go"
export PATH="$GOPATH/bin:$PATH"
```

Building `gopherclient` however can be done anywhere; e.g:

```#!sh
mkdir -p $HOME/Contributions
cd $HOME/Contributions
git clone https://github.com/prologic/gopherclient/issues/14
cd gopherclient
make
```

Try this :)

**Comment by prologic** [comment]
I am attempting to put together a `Dockerfile.opensuse` image that will build this for openSUSE.

So far so good:

```#!sh
# pkg-config --cflags  -- gtk+-3.0 webkit2gtk-4.0
Package gtk+-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtk+-3.0.pc'
to the PKG_CONFIG_PATH environment variable
Package 'gtk+-3.0', required by 'virtual:world', not found
Package 'webkit2gtk-4.0', required by 'virtual:world', not found
pkg-config: exit status 1
make: *** [Makefile:17: deps] Error 2
```

**Comment by prologic** [comment]
Success!

```#!sh
sh-5.0# make build
go: finding github.com/GeertJohan/go.rice/rice latest
# github.com/zserge/webview
In file included from webview.go:27,
                 from _cgo_export.c:4:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h: In function 'external_message_received_cb':
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:253:3: warning: 'webkit_javascript_result_get_global_context' is deprecated [-Wdeprecated-declarations]
  253 |   JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
      |   ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
   49 | webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from webview.go:27,
                 from _cgo_export.c:4:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:254:3: warning: 'webkit_javascript_result_get_value' is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
  254 |   JSValueRef value = webkit_javascript_result_get_value(r);
      |   ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
   52 | webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# github.com/zserge/webview
In file included from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h: In function 'external_message_received_cb':
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:253:3: warning: 'webkit_javascript_result_get_global_context' is deprecated [-Wdeprecated-declarations]
  253 |   JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
      |   ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
   49 | webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:254:3: warning: 'webkit_javascript_result_get_value' is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
  254 |   JSValueRef value = webkit_javascript_result_get_value(r);
      |   ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
   52 | webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 -> Building latest-dev
# github.com/zserge/webview
In file included from webview.go:27,
                 from _cgo_export.c:4:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h: In function 'external_message_received_cb':
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:253:3: warning: 'webkit_javascript_result_get_global_context' is deprecated [-Wdeprecated-declarations]
  253 |   JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
      |   ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
   49 | webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from webview.go:27,
                 from _cgo_export.c:4:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:254:3: warning: 'webkit_javascript_result_get_value' is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
  254 |   JSValueRef value = webkit_javascript_result_get_value(r);
      |   ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from webview.go:27,
                 from _cgo_export.c:4:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
   52 | webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# github.com/zserge/webview
In file included from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h: In function 'external_message_received_cb':
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:253:3: warning: 'webkit_javascript_result_get_global_context' is deprecated [-Wdeprecated-declarations]
  253 |   JSGlobalContextRef context = webkit_javascript_result_get_global_context(r);
      |   ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:49:1: note: declared here
   49 | webkit_javascript_result_get_global_context (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:254:3: warning: 'webkit_javascript_result_get_value' is deprecated: Use 'webkit_javascript_result_get_js_value' instead [-Wdeprecated-declarations]
  254 |   JSValueRef value = webkit_javascript_result_get_value(r);
      |   ^~~~~~~~~~
In file included from /usr/include/webkitgtk-4.0/webkit2/webkit2.h:56,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.h:44,
                 from /go/pkg/mod/github.com/zserge/webview@v0.0.0-20181018084947-f390a2df9ec5/webview.go:27:
/usr/include/webkitgtk-4.0/webkit2/WebKitJavascriptResult.h:52:1: note: declared here
   52 | webkit_javascript_result_get_value          (WebKitJavascriptResult *js_result);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Built gopherclient v0.1.0-dev (HEAD)
sh-5.0# ls -1
Dockerfile
Dockerfile.opensuse
LICENSE
Makefile
README.md
_config.yml
assets
go.mod
go.sum
gopherclient
main.go
rice-box.go
screenshot.png
template.go
version.go
sh-5.0# ./gopherclient -h
Usage of ./gopherclient:
  -v=false: display version information
sh-5.0#
```

I'll commit this `Dockerfile.opensuse`.

You should see it in master shortly.

Use the `RUN` instructions therein as a "guide" to building on your own system

**Comment by prologic** [comment]
I'm _hoping_ you succeed in building a binary for openSUSE now :) If you run into any different issues; please open another issue :)
